utilities/PerformanceTesting/query.py

```diff
@@ -0,0 +1,292 @@
+    needed to query the Prometheus API
+
+    :return: Json containing information about Collector counters
+    """
+    res = {}
+    counters = ["net_conn_deltas", "net_conn_updates", "net_conn_inactive"]
+
+    for counter in counters:
+        for stat in g_stat_names:
+            query = f'{stat} by (job) '
+            query += f'(rox_collector_counters{{type="{counter}"}})'
+            query_name = f'{g_stat_names[stat]} of {counter} over pods'
+
+            res[query_name] = querier.query_and_get_value(query)
+
+    return res
+
+
+def get_sensor_network_flows(querier, query_window):
+    """
+    Performs Prometheus queries relevant to sensor network flows
+
+    :param querier: An object containing information
+    needed to query the Prometheus API
+    :param query_window: The window over which rates are computed by Prometheus
+
+    :return: Json containing information about sensor network flows
+    """
+    res = {}
+    for stat in g_stat_names:
+        for flow_type in ["incoming", "outgoing"]:
+            for protocol in ["L4_PROTOCOL_TCP", "L4_PROTOCOL_UDP"]:
+                metric = 'rox_sensor_network_flow_total_per_node'
+                query = f'{stat} by (job)  ({metric}'
+                query += f'{{Protocol="{protocol}",Type="{flow_type}"}})'
+                query_name = f'{g_stat_names[stat]} number of {flow_type}'
+                query_name += f' {protocol} network flows over pods'
+
+                res[query_name] = querier.query_and_get_value(query)
+
+    metrics = [
+                "rox_sensor_network_flow_host_connections_added",
+                "rox_sensor_network_flow_host_connections_removed",
+                "rox_sensor_network_flow_external_flows"
+            ]
+
+    for metric in metrics:
+        query = metric
+        rate_query = f'rate({metric}[{query_window}])'
+
+        query_name = metric
+        rate_query_name = 'Average {metric} per second'
+
+        res[query_name] = querier.query_and_get_value(query)
+        res[rate_query_name] = querier.query_and_get_value(rate_query)
+
+    return res
+
+
+def get_cpu_mem_network_usage(querier, query_window):
+    """
+    Performs Prometheus queries relevant to cpu, memory, and network IO usage
+
+    :param querier: An object containing information
+    needed to query the Prometheus API
+    :param query_window: The window over which rates are computed by Prometheus
+
+    :return: Json containing information about cpu, memory,
+    and network IO usage
+    """
+    res = {}
+    metric_names = {
+            'container_network_receive_bytes_total':
+            'network received (bytes per second)',
+            'container_network_transmit_bytes_total':
+            'network transmited (bytes per second)'
+            }
+
+    for stat in g_stat_names:
+        cpu_query = f'{stat} by (job) (rate(container_cpu_usage_seconds_total'
+        cpu_query += f'{{namespace="stackrox"}}[{query_window}]) * 100)'
```

Comment:
appreciate that we are now using fstrings but i think we can go a step further, to avoid string concatenation altogether.

```suggestion
        cpu_query = f'{stat} by (job) (rate(container_cpu_usage_seconds_total{{namespace="stackrox"}}[{query_window}]) * 100)'
```

Reply:
Done